docs/CHANGELOG.md

```diff
@@ -13,6 +13,29 @@ Changes are grouped as follows
 - `Fixed` for any bug fixes.
 - `Security` in case of vulnerabilities.
 
+## [0.99.25] - 07-06-24
+### Added
+- `create_external_id_factory` in `external_id_factories` which takes in a configuration to produce external_id's of the
+  format `prefix:suffix`
+- `ExternalIdFactory` class as a template for custom functions to define for `DataModelWrite.external_id_factory`
+- Helper functions for unique external_id generation
+  - `domain_name` returns a clean string from the domain class name
+  - `uuid` returns a uuid
+  - `sha256` returns a sha256 has of the data
+  - `incremental_id` returns an incremental integer based on domain class
+- `incremental_id_factory` an instance of the `ExternalIdFactory` that implements the `incremental_id` helper
+
+### Changed
+- More flexible external_id_factory functions
+  - `sha256_factory` is now an instance of the `ExternalIdFactory` class
+  - `uuid_factory` is now an instance of the `ExternalIdFactory` class
+
+### Removed
+- The following functions in `external_id_factories` have been removed and replaced
+  - `create_incremental_factory` is replaced by `create_external_id_factory(suffix_ext_id_factory=incremental_id_factory)`
+  - `create_sha256_factory` is replaced by `create_external_id_factory(suffix_ext_id_factory=sha256_factory)`
+  - `create_uuid_factory` is replaced by `create_external_id_factory()`
```

Comment:
Currently, pygen is committed to not do breaking changes, so we need to add methods like this to the `external_id_factory`. Furthermore, shouldn't it be create_external_id_factory(suffix_ext_id_factory=sha256_factory())` or am I missing something?
```python
def create_sha256_factory(shorten: bool = True) -> Callable[[type, dict], str]:
    """
    This creates a sha256 hash external id factory.

    Args:
        shorten: If True, the external id will be shortened to 7 characters.

    Returns:
        A factory function that can be set on the DomainModelApply class.
    """
    arg = "7" if shorten else ""
    warnings.warn(
        "create_sha256_factory is deprecated and will be removed in v1.0. "
        f"Use create_external_id_factory(suffix_ext_id_factory=sha256_factory({arg})) instead.",
        DeprecationWarning,
        stacklevel=2,
    )
    return create_external_id_factory(suffix_ext_id_factory=sha256_factory(7 if shorten else None))
```

Reply:
Added them back with deprecation warning